.github/workflows/ci-aarch64.yml

```diff
@@ -0,0 +1,100 @@
+# *******************************************************************************
+# Copyright 2024 Arm Limited and affiliates.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# *******************************************************************************
+
+name: "CI AArch64"
+
+#* To avoid duplicate jobs running when both push and PR is satisfied, we use this:
+#* https://github.com/orgs/community/discussions/26940#discussioncomment-5686753
+on:
+  push:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+#* Stop stale workflows when pull requests are updated: https://stackoverflow.com/a/70972844
+#* Does not apply to the main branch.
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+# Declare default permissions as read only.
+permissions: read-all
+
+jobs:
+  macos:
+    name: macOS
+    runs-on: macos-14
+    strategy:
+      matrix:
+        compiler: [
+          {CC: clang, CXX: clang++},
+          {CC: gcc-14, CXX: g++-14}
+        ]
+        config: [
+          {CMAKE_BUILD_TYPE: Debug, ACL_WITH_ASSERTS: '1'},
+          {CMAKE_BUILD_TYPE: Release, ACL_WITH_ASSERTS: '0'}
+        ]
+    steps:
```

Comment:
In the future it would be great to move out as much logic as possible from actions to build.sh/test.sh scripts like it's [done for Azure CI](https://github.com/oneapi-src/oneDNN/blob/main/.github/automation/.azure-pipeline.yml). This would help with with debugging and reusability of the code.